VungleAdapter/src/main/java/com/chartboost/mediation/vungleadapter/VungleAdapter.kt

```diff
@@ -664,153 +524,198 @@ class VungleAdapter : PartnerAdapter {
-                    }
+            fun loadVungleFullScreenAd(ad: () -> BaseFullscreenAd) {
+                ad().apply {
+                    adListener = createFullScreenAdListener(request, listener, continuation)
+                    load(adm)
+                }
+            }
 
-                    override fun onError(placementId: String?, exception: VungleException?) {
-                        PartnerLogController.log(
-                            LOAD_FAILED,
-                            "Placement: $placementId. Error code: ${exception?.exceptionCode}. " +
-                                    "Message: ${exception?.message}"
-                        )
+            when (request.format) {
+                AdFormat.INTERSTITIAL -> loadVungleFullScreenAd {
+                    InterstitialAd(context, request.partnerPlacement, adConfig)
+                }
+                AdFormat.REWARDED -> loadVungleFullScreenAd {
+                    RewardedAd(context, request.partnerPlacement, adConfig)
+                }
+                else -> {
+                    if (request.format.key == "rewarded_interstitial") {
+                        loadVungleFullScreenAd {
+                            RewardedAd(context, request.partnerPlacement, adConfig)
+                        }
+                    } else {
+                        PartnerLogController.log(LOAD_FAILED)
                         resumeOnce(
                             Result.failure(
                                 ChartboostMediationAdException(
-                                    getChartboostMediationError(
-                                        exception
-                                    )
+                                    ChartboostMediationError.CM_LOAD_FAILURE_UNSUPPORTED_AD_FORMAT
                                 )
                             )
                         )
                     }
                 }
-            )
+            }
         }
     }
 
+
     /**
      * Attempt to show a Vungle fullscreen ad.
      *
      * @param partnerAd The [PartnerAd] object containing the Vungle ad to be shown.
-     * @param listener A [PartnerAdListener] to notify Chartboost Mediation of ad events.
+     *
+     * @return Result.success(PartnerAd) if the ad was successfully shown, Result.failure(Exception) otherwise.
      */
-    private suspend fun showFullscreenAd(
-        partnerAd: PartnerAd,
-        listener: PartnerAdListener?
+    private fun showFullscreenAd(
+        partnerAd: PartnerAd
     ): Result<PartnerAd> {
-        val adm = adms.remove(partnerAd.request.partnerPlacement)
-
-        return suspendCancellableCoroutine { continuation ->
-            fun resumeOnce(result: Result<PartnerAd>) {
-                if (continuation.isActive) {
-                    continuation.resume(result)
-                }
+        fun canPlay(canPlay: () -> Boolean, play: () -> Unit): Result<PartnerAd> {
+            return if (canPlay()) {
```

Comment:
function name shadowed? no compiler warning about this?

Reply:
Nope. No warning. 😃

Reply:
But removed the method based on this [comment](https://github.com/ChartBoost/chartboost-mediation-android-adapter-vungle/pull/43#discussion_r1326471724).